plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -121,18 +142,20 @@ func TestMetricsServerError(t *testing.T) {
 	defer mockPrometheus.Close()
 
 	logger := zap.NewNop()
+	tracer, exp, closer := tracerProvider(t)
+	defer closer()
 	address := mockPrometheus.Listener.Addr().String()
-	reader, err := NewMetricsReader(logger, config.Configuration{
+	reader, err := NewMetricsReader(config.Configuration{
 		ServerURL:      "http://" + address,
 		ConnectTimeout: defaultTimeout,
-	})
+	}, logger, tracer)
 	require.NoError(t, err)
-
 	m, err := reader.GetCallRates(context.Background(), &params)
 	assert.NotNil(t, m)
-
 	require.Error(t, err)
 	assert.Contains(t, err.Error(), "failed executing metrics query")
+	require.Len(t, exp.GetSpans(), 1, "HTTP request was traced and span reported")
+	assert.Equal(t, exp.GetSpans()[0].Name, "service_call_rate")
```

Comment:
please read the signature of Equal() more carefully and find a mistake

Reply:
so silly. on it.